c7n/resources/s3.py

```diff
@@ -2582,7 +2582,7 @@ def process_bucket(self, bucket):
         s3 = bucket_client(local_session(self.manager.session_factory), bucket)
 
         # Adjust the existing lifecycle by adding/deleting/overwriting rules as necessary
-        config = (bucket['Lifecycle'] or {}).get('Rules', [])
+        config = (bucket.get('Lifecycle') or {}).get('Rules', [])
```

Comment:
This is necessary because in the S3 augment step, if the permission fails for a given method then it doesn't set the key.  I wonder if it should set the default instead.

Reply:
we should probably filter out buckets before processing if we don't have access to get lifecycle configuration, it will be listed in an an annotation `deniedmethods` (sp?) per the bucket augment.

Reply:
Yes - good point.  Fixed.

It would be odd to have permissions allowing a person to put the lifecycle but not get it.  But if that does happen this would end up deleting the previous lifecycle.